index.js

```diff
@@ -172,77 +81,150 @@ export default class Purchases {
     return false;
   }
 
+  /**
+   * Enum for attribution networks
+   * @readonly
+   * @enum {Number}
+   */
   static ATTRIBUTION_NETWORKS = {
     APPLE_SEARCH_ADS: 0,
     ADJUST: 1,
     APPSFLYER: 2,
-    BRANCH: 3
+    BRANCH: 3,
+    TENJIN: 4
   };
 
-  /** 
-    Add a dict of attribution information
-    @param data Attribution data from AppsFlyer, Adjust, or Branch
-    @param network Which network, see Purchases.ATTRIBUTION_NETWORKS
-  */
+  /**
+   * Add a dict of attribution information
+   * @param {Dict} data Attribution data from AppsFlyer, Adjust, or Branch
+   * @param {ATTRIBUTION_NETWORKS} network Which network, see Purchases.ATTRIBUTION_NETWORKS
+   */
   static addAttributionData(data, network) {
     RNPurchases.addAttributionData(data, network);
   }
 
-  /** Gets the map of entitlements -> offerings -> products
-    @returns {Promise<Map<String, Map<String, Product>>>} Promise of entitlements structure
-  */
+  /**
+   * Gets the map of entitlements -> offerings -> products
+   * @returns {Promise<Map<String, Map<String, Product>>>} Promise of entitlements structure
+   */
   static getEntitlements() {
     return RNPurchases.getEntitlements();
   }
 
-  /** Fetch the product info.
-      @param {[String]} productIdentifiers Array of product identifiers
-      @returns {Promise<Array>} A promise of product arrays
-  */
+  /**
+   * Fetch the product info
+   * @param {[String]} productIdentifiers Array of product identifiers
+   * @param {String} type Optional type of products to fetch, can be inapp or subs. Subs by default
+   * @returns {Promise<Array>} A promise of product arrays
+   */
   static getProducts(productIdentifiers, type = "subs") {
     return RNPurchases.getProductInfo(productIdentifiers, type);
   }
 
-  /** Make a purchase
-      @param {String} productIdentifier The product identifier of the product you want to purchase.
-  */
+  /**
+   * Make a purchase
+   * @param {String} productIdentifier The product identifier of the product you want to purchase.
+   * @param {Array<String>} oldSKUs Optional array of skus you wish to upgrade from.
+   * @param {String} type Optional type of product, can be inapp or subs. Subs by default
+   * @returns {Promise<Object>} A promise of purchaser info object and a boolean indicating if user cancelled
```

Comment:
Hmm, docs seem a little strange.

Seems like an `Either<>` or `Result` type would be useful here.